krakend/content.md

```diff
@@ -0,0 +1,150 @@
+
+#### Checking the syntax of your configuration file
+
+See the [check command](https://www.krakend.io/docs/commands/check/)
+
+```console
+    docker run -it -v $PWD:/etc/krakend/ %%IMAGE%% check --config krakend.json
+```
+
+#### Show the help:
+
+	docker run -it %%IMAGE%% help
+
+### Building your custom KrakenD image
+
+Most production deployments will not want to rely on mounting a volume for the container but to use their image based on `%%IMAGE%%`:
+
+Your `Dockerfile` could look like this:
+
+```Dockerfile
+FROM %%IMAGE%%:<version>
+# NOTE: Avoid using :latest image on production. Stick to a major version instead.
+
+COPY krakend.json /etc/krakend/krakend.json
+
+# Check and test that the file is valid
+RUN krakend check -d -t -c /etc/krakend/krakend.json
+```
+
+If you want to manage your KrakenD configuration using multiple files and folders, reusing templates, and distributing the configuration amongst your teams, you can use the [flexible configuration (FC)](https://www.krakend.io/docs/configuration/flexible-config/). The following `Dockerfile` combines FC, the `krakend check` command, and a 2-step build.
+
+```Dockerfile
+FROM %%IMAGE%%:<version> as builder
+
+COPY krakend.tmpl .
+COPY config .
+
+# Save temporary output file to /tmp to avoid permission errors
+RUN FC_ENABLE=1 \
+    FC_OUT=/tmp/krakend.json \
+    FC_PARTIALS="/etc/krakend/partials" \
+    FC_SETTINGS="/etc/krakend/settings" \
+    FC_TEMPLATES="/etc/krakend/templates" \
+    krakend check -d -t -c krakend.tmpl
+
+# Copy the output file only and discard any other files
+FROM %%IMAGE%%:<version>
+COPY --from=builder /tmp/krakend.json .
+```
+
+Then build with `docker build -t my_krakend .`
+
+The configuration above assumes you have a folder structure like the following:
+
+	.
+	├── config
+	│   ├── partials
+	│   ├── settings
+	│   │   └── env.json
+	│   └── templates
+	│       └── some.tmpl
+	├── Dockerfile
+	└── krakend.tmpl
+
+### Docker Compose example
+
+Finally, a simple `docker-compose` file to start KrakenD with your API would be:
+
+```yaml
+version: "3"
+services:
+  krakend:
+    image: %%IMAGE%%:<version>
+    ports:
+      - "8080:8080"
+    volumes:
+      - ./:/etc/krakend
+```
+
+And another one that uses the flexible configuration and a custom template filename (`my_krakend.tmpl`) on each start:
+
+```yaml
+version: "3"
+services:
+  krakend:
+    image: %%IMAGE%%:<version>
+    ports:
+      - "8080:8080"
+    volumes:
+      - ./:/etc/krakend
+    environment:
+      - FC_ENABLE=1
+      - FC_OUT=/tmp/krakend.json
+      - FC_PARTIALS="/etc/krakend/config/partials"
+      - FC_SETTINGS="/etc/krakend/config/settings/prod"
+      - FC_TEMPLATES="/etc/krakend/config/templates"
+    command:
+      command: ["krakend", "run", "-c", "krakend.tmpl", "-d"]
+```
+
+### Container permissions and commands
+
+All `krakend` commands are executed as `krakend` user (uid=1000) through `su-exec`, and the rest of the commands (e.g., `sh`) are executed as root.
```

Comment:
As noted in the other review, I would suggest not marrying yourself explicitly to `su-exec`:

```suggestion
All `krakend` commands are executed as `krakend` user (uid=1000), and the rest of the commands (e.g., `sh`) are executed as root.
```